pkg/client/connector/intercept.go

```diff
@@ -214,13 +214,18 @@ func (tm *trafficManager) addIntercept(c context.Context, ir *rpc.CreateIntercep
 		}()
 	}
 
+	apiKey, err := tm.getAPIKey(c, "agent", false)
```

Comment:
Consider including the agent name (e.g. `"agent-"+mechanism` or something) in this string?